firefox-ios/Client/Application/WindowManager.swift

```diff
@@ -196,7 +199,30 @@ final class WindowManagerImplementation: WindowManager, WindowTabsSyncCoordinato
                    level: .debug,
                    category: .window)
 
-        let result = nextWindowUUIDToOpen(filteredUUIDs)
+        // On iPhone devices, we expect there only to ever be a single window. If there
+        // are >1 windows we've encountered some type of unexpected state.
+        let isIpad = (UIDevice.current.userInterfaceIdiom == .pad)
+
+        let result: (uuid: WindowUUID, isNew: Bool)
+        if !isIpad {
+            // if onDiskUUIDs.count > 1 { ... }
+            // TODO: [FXIOS-9544] If >1 tab session files, clean up & merge
```

Comment:
cc @OrlaM I believe this is where we would want to perform the clean-up and merge of files on iPhone